src/Amqp/JobQueue/AmqpWorker.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Icecave\Overpass\Amqp\JobQueue;
+
+use Eloquent\Asplode\Error\ErrorException;
+use Error;
+use Exception;
+use Icecave\Overpass\Amqp\ChannelDispatcher;
+use Icecave\Overpass\JobQueue\Exception\InvalidMessageException;
+use Icecave\Overpass\JobQueue\Message\MessageSerialization;
+use Icecave\Overpass\JobQueue\Message\MessageSerializationInterface;
+use Icecave\Overpass\JobQueue\Request;
+use Icecave\Overpass\JobQueue\WorkerInterface;
+use Icecave\Overpass\Serialization\JsonSerialization;
+use Icecave\Overpass\Serialization\SerializationInterface;
+use LogicException;
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use ReflectionClass;
+use Icecave\Overpass\JobQueue\Exception\DiscardException;
+
+class AmqpWorker implements WorkerInterface
+{
+    use LoggerAwareTrait;
+
+    /**
+     * @param LoggerInterface             $logger
+     * @param AMQPChannel                 $channel
+     * @param DeclarationManager|null     $declarationManager
+     * @param SerializationInterface|null $serialization
+     * @param ChannelDispatcher           $channelDispatcher
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        AMQPChannel $channel,
+        DeclarationManager $declarationManager = null,
+        MessageSerializationInterface $serialization = null,
+        ChannelDispatcher $channelDispatcher = null
+    ) {
+        $this->channel = $channel;
+        $this->declarationManager = $declarationManager ?: new DeclarationManager($channel);
+        $this->serialization = $serialization ?: new MessageSerialization(new JsonSerialization());
+        $this->channelDispatcher = $channelDispatcher ?: new ChannelDispatcher();
+        $this->jobs = [];
+        $this->consumerTags = [];
+
+        $this->setLogger($logger);
+    }
+
+    /**
+     * Expose a job.
+     *
+     * @param string   $name The public name of the job
+     * @param callable $job  The job implementation
+     *
+     * @throws LogicException if the worker is already running
+     */
+    public function expose($name, callable $job)
```

Comment:
`expose` doesn't feel right outside of the RPC context.  Also, remember that a job is an *instance* of a specific task.  What do you think about:

`register($name, callable $handler)` - So you register a handler against a specific task name. 

The terminology of `QueueInterface::enqueue()` looked right.